### PR TITLE
PRESS2 305 | Build Fonts Design Step

### PR DESCRIPTION
**Pull request by officiallygod**

### CODE REVIEW

File: includes/Data/Fonts.php [hunk 1]
```diff
@@ -0,0 +1,229 @@
+<?php
```
Comment (arunshenoy99) [inline includes/Data/Fonts.php:1]: What do you think of a structure where `Colors.php` and `Fonts.php` go into `Themes.php` or a folder called `Themes` similar to controllers that contains all 3 files. As we give more customisation for a theme we will have more of these in the parent folder.

File: includes/Data/Fonts.php [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+
+          if (!isset($pattern_slugs)) {
+               return new \WP_Error(
+                    'Theme Fonts not found',
+                    'No WordPress Fonts are available for this theme.',
+                    array('status' => 404)
+               );
+          }
+
+          return $pattern_slugs;
```
Comment (arunshenoy99) [inline includes/Data/Fonts.php:227]:
```suggestion
          return isset(self::$theme_fonts[$active_theme]) ? self::$theme_fonts[$active_theme] : false;
```

File: includes/RestApi/Themes/ThemeFontsController.php [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+     /**
+      * Retrieves the active theme font variations.
+      *
+      * @return array|\WP_Error
+      */
+     public function get_theme_fonts()
+     {
+          $theme_font_palettes = Fonts::get_fonts_from_theme();
+          return $theme_font_palettes;
```
Comment (arunshenoy99) [inline includes/RestApi/Themes/ThemeFontsController.php:62]: Let's return the `WP_Error` here if it is false

File: src/OnboardingSPA/components/Drawer/DrawerPanel/DesignTypography.js [hunk 4]
```diff
@@ -1,17 +1,174 @@
...
-					'wp-module-onboarding'
-				)}
-			</p>
+		<div ref={drawerFontOptions} className='theme-fonts--drawer'>
+			<h2>{__('Font Palettes', 'wp-module-onboarding')}</h2>
+			{/* { selectedFont && 
+				<div className='theme-fonts--drawer--reset' onClick={resetFonts}>
+					<div>Reset Button</div>
+				</div>
+			} */}
```
Comment (arunshenoy99) [inline src/OnboardingSPA/components/Drawer/DrawerPanel/DesignTypography.js:165]: Need to discuss this as well right?

**Review by arunshenoy99 — COMMENTED** [review]
Let's run php and javascript formatters

File: includes/Data/Fonts.php (hunk 1 above)
Comment (officiallygod) [reply includes/Data/Fonts.php:1]: Makes sense, I'll look into that.

File: src/OnboardingSPA/components/Drawer/DrawerPanel/DesignTypography.js (hunk 4 above)
Comment (officiallygod) [reply src/OnboardingSPA/components/Drawer/DrawerPanel/DesignTypography.js:165]: Yes, We need to. Thanks

**Review by arunshenoy99 — APPROVED** [review]

File: src/OnboardingSPA/components/Drawer/DrawerPanel/DesignTypography.js [hunk 5]
```diff
@@ -1,17 +1,191 @@
...
+
+		// Add the font CSS changes to Preview Settings
+		storedPreviewSettings.settings.styles = [
+			...result,
+			{
+				id: 'customFontProperty',
+				css: `body {font-family: ${fontPalettesCopy[fontStyle]?.styles?.typography?.fontFamily} !important; }
+						h6 { font-family: ${fontPalettesCopy[fontStyle]?.styles?.blocks['core/heading']?.typography?.fontFamily} !important; }`,
+			},
+		]
```
Comment (arunshenoy99) [inline src/OnboardingSPA/components/Drawer/DrawerPanel/DesignTypography.js:98]: Need to remove this hardcoding as part of the 6.1 fix story

**Pull request merged**